pyaedt/edb_core/components.py

```diff
@@ -379,6 +380,99 @@ def get_components_from_nets(self, netlist=None):
                 cmp_list.append(refdes)
         return cmp_list
 
+    @aedt_exception_handler
+    def _get_edb_pin_from_pin_name(self, cmp, pin):
+        if not isinstance(cmp, self._edb.Cell.Hierarchy.Component):
+            return False
+        if not isinstance(pin, str):
+            pin = pin.GetName()
+        pins = self.get_pin_from_component(component=cmp, pinName=pin)
+        if pins:
+            return pins[0]
+        return False
+
+    @aedt_exception_handler
+    def get_component_placement_vector(
+        self,
+        mounted_component,
+        hosting_component,
+        mounted_component_pin1=None,
+        mounted_component_pin2=None,
+        hosting_component_pin1=None,
+        hosting_component_pin2=None,
+    ):
+        """Get the placement vector between 2 components.
+
+        Parameters
+        ----------
+        mounted_component : str
+            Mounted component name.
+        hosting_component : str
+            Hosting component name.
+        mounted_component_pin1 : str
+            Mounted component Pin 1 name.
+        mounted_component_pin2 : str
+            Mounted component Pin 2 name.
+        hosting_component_pin1 : str
+            Hosted component Pin 1 name.
+        hosting_component_pin2 : str
+            Hosted component Pin 2 name.
+
+        Returns
+        -------
+        tuple
+            Tuple of Vector offset, rotation and solder height.
+
+        Examples
+        --------
+        >>> edb1 = Edb(edbpath=targetfile1,  edbversion="2021.2")
+        >>> hosting_cmp = edb1.core_components.get_component_by_name("U100")
+        >>> mounted_cmp = edb2.core_components.get_component_by_name("BGA")
+        >>> vector, rotation, solder_ball_height = edb1.core_components.get_component_placement_vector(
+        >>>                                             mounted_component=mounted_cmp,
+        >>>                                             hosting_component=hosting_cmp,
+        >>>                                             mounted_component_pin1="A12",
+        >>>                                             mounted_component_pin2="A14",
+        >>>                                             hosting_component_pin1="A12",
+        >>>                                             hosting_component_pin2="A14")
```

Comment:
Nitpick: should use `...` when continuing a command going over multiple lines.

```suggestion
        ...                                             mounted_component=mounted_cmp,
        ...                                             hosting_component=hosting_cmp,
        ...                                             mounted_component_pin1="A12",
        ...                                             mounted_component_pin2="A14",
        ...                                             hosting_component_pin1="A12",
        ...                                             hosting_component_pin2="A14")
```